src/main/java/com/dev/cinema/security/CustomUserDetailService.java

```diff
@@ -0,0 +1,38 @@
+package com.dev.cinema.security;
+
+import com.dev.cinema.model.User;
+import com.dev.cinema.service.UserService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.userdetails.User.UserBuilder;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+
+@Service
+public class CustomUserDetailService implements UserDetailsService {
+    private final UserService userService;
+
+    @Autowired
+    public CustomUserDetailService(UserService userService) {
+        this.userService = userService;
+    }
+
+    @Override
+    public UserDetails loadUserByUsername(String username) {
+        User user = userService.findByEmail(username).orElseThrow();
```

Comment:
let's throw UsernameNotFoundException in the `orElseThrow()` method)

Reply:
Changed.